File: PoketDex/app/src/main/java/pt/rfsfernandes/data/repository/ResponseCallBack.java
package pt.rfsfernandes.data.repository;

/**
 * Interface ResponseCallBack created at 1/16/21 15:54 for the project PoketDex
 * By: rodrigofernandes
 */
public interface ResponseCallBack<T>{
  void onSuccess(T response);
  void onFailure(String errorMessage);
}

File: PocketDex/app/src/main/java/pt/rfsfernandes/pocketdex/custom/callbacks/ResponseCallBack.kt
package pt.rfsfernandes.pocketdex.custom.callbacks

/**
 * Interface ResponseCallBack created at 1/16/21 15:54 for the project PoketDex
 * By: rodrigofernandes
 */
interface ResponseCallBack<T> {
    fun onSuccess(response: T)
    fun onFailure(errorMessage: String?)
}